src/prpy/planning/ompl.py

```diff
@@ -163,67 +167,115 @@ def _Plan(self, robot, goal=None, tsrchains=None, timeout=30., shortcut_timeout=
+    default value is calculated from the active DOFs of 'robot' such that there
+    will be an fixed number of state validity checks per extension. That number
+    is configured in the constructor of this class.
+
+    @param robot with active DOFs set
+    @param ompl_args arguments to append to, defaults to an empty dictionary
+    @return copy of ompl_args with 'range' set to the default value
+    """
     def _SetPlannerRange(self, robot, ompl_args=None):
-        from copy import deepcopy
+        epsilon = 1e-6
 
         if ompl_args is None:
             ompl_args = dict()
         else:
             ompl_args = deepcopy(ompl_args)
 
-        # Set the default range to the collision checking resolution.
         if 'range' not in ompl_args:
-            epsilon = 1e-6
-
-            # Compute the collision checking resolution as a conservative
-            # fraction of the state space extents. This mimics the logic inside
-            # or_ompl used to set collision checking resolution.
+            # Compute the maximum DOF range, treating circle joints as SO(2).
             dof_limit_lower, dof_limit_upper = robot.GetActiveDOFLimits()
-            dof_ranges = dof_limit_upper - dof_limit_lower
+            dof_ranges = numpy.zeros(robot.GetActiveDOF())
+
+            for index, dof_index in enumerate(robot.GetActiveDOFIndices()):
+                joint = robot.GetJointFromDOFIndex(dof_index)
+                axis_index = dof_index - joint.GetDOFIndex()
+
+                if joint.IsCircular(axis_index):
+                    dof_ranges[index] = 2 * numpy.pi
+                else:
+                    dof_ranges[index] = (dof_limit_upper[index]
+                                       - dof_limit_lower[index])
+
+            # Duplicate the logic in or_ompl to convert the DOF resolutions to
+            # a fraction of the longest extent of the state space.
+            maximum_extent = numpy.max(dof_ranges)
             dof_resolution = robot.GetActiveDOFResolutions()
-            ratios = dof_resolution / dof_ranges
-            conservative_ratio = numpy.min(ratios)
-
-            # Convert the ratio back to a collision checking resolution. Set
-            # RRT-Connect's range to the same value used for collision
-            # detection inside OpenRAVE.
-            longest_extent = numpy.max(dof_ranges)
-            ompl_range = conservative_ratio * longest_extent - epsilon
-
-            ompl_args['range'] = ompl_range
-            logger.debug('Defaulted RRT-Connect range parameter to %.3f.',
-                         ompl_range)
+            conservative_resolution = numpy.min(dof_resolution)
+            conservative_fraction = conservative_resolution / maximum_extent
+
+            if self.multiplier is not None:
+                multiplier = self.multiplier
+            elif self.fraction is not None:
+                multiplier = int(self.fraction / conservative_fraction + 0.5)
```

Comment:
The C side of me likes this, but perhaps we can just use Python's `round` function here?